### PR TITLE
Replace p2p library

### PR DESCRIPTION
**Pull request by jjyr**

### CODE REVIEW

**Comment by driftluo** [comment]
This pr has the same problem as cita's (pr)[https://github.com/cryptape/cita/pull/278].

Doesn't notice the definition of `name`/`support_version`. Used the default definition, which is `/p2p/{protocol_id}`.

In fact, the protocol id is just an internal index, so there is no need to expose it externally. The same protocol for each service can be implemented internally as a different id

File: network/src/network.rs [hunk 1]
```diff
@@ -678,82 +435,106 @@ impl Network {
...
+                    return Err(Error::Io(err));
                 }
             };
         }
 
         // dial reserved nodes and bootnodes
         {
             let network = Arc::clone(&network);
-            let dial_timeout = Duration::from_secs(DIAL_BOOTNODE_TIMEOUT);
+            let _dial_timeout = Duration::from_secs(DIAL_BOOTNODE_TIMEOUT);
```
Comment (quake) [inline network/src/network.rs]: unused variable?

File: network/src/network.rs [hunk 2]
```diff
@@ -240,265 +246,110 @@ impl Network {
...
             }
-            Err(err) => Err(err),
+            SessionType::Server => {
+                peers_registry.accept_inbound_peer(peer_id.clone(), connected_addr, session)
+            }
+        }?;
+        // add session to peer
+        match peers_registry.get_mut(&peer_id) {
+            Some(peer) => match peer.protocol_version(protocol_id) {
+                Some(_) => return Err(ProtocolError::Duplicate(protocol_id).into()),
```
[the remark this answers is not in the document]
Comment (jjyr) [reply network/src/network.rs]: Yes, use HashMap is better.

**Comment by quake** [comment]
This PR breaks integration test, could you help to debug?

```shell
make build
cd test
cargo run
```

error
```
thread 'main' panicked at 'assertion failed: node0.rpc_client().get_block(hash.clone()).call().unwrap().is_some()', test/src/specs/block_relay.rs:20:9
```

it looks like node cannot connect to peers.

**Comment by jjyr** [comment]
> This PR breaks integration test, could you help to debug?
> 
> ```shell
> make build
> cd test
> cargo run
> ```
> 
> error
> 
> ```
> thread 'main' panicked at 'assertion failed: node0.rpc_client().get_block(hash.clone()).call().unwrap().is_some()', test/src/specs/block_relay.rs:20:9
> ```
> it looks like node cannot connect to peers.

Fixed in last commit.

**Pull request merged**